inc/class-importer.php

```diff
@@ -281,11 +281,20 @@ protected function update_post_from_markdown_source( $post_id ) {
 		$etag = wp_remote_retrieve_header( $response, 'etag' );
 
 		$markdown = wp_remote_retrieve_body( $response );
-		// Strip YAML doc from the header
-		$markdown = preg_replace( '#^---(.+)---#Us', '', $markdown );
+
+		// Get YAML doc from the header
+		preg_match( '#^---(.+)---#Us', $markdown, $yaml );
+		if ( $yaml ) {
+			$yaml_parser = new Yaml();
+			$yaml        = $yaml_parser->loadString( $yaml[1] );
+			// Strip YAML doc from the header
+			$markdown = preg_replace( '#^---(.+)---#Us', '', $markdown );
```

Comment:
This could use `substr( $markdown, strlen( $yaml[0] ) )` (I'd rename `$yaml` to `$yaml_matches` though).